### PR TITLE
Change some flags (breaking)

### PR DESCRIPTION
**Pull request by thockin**
* Make the --root flag required, no default (breaking)
* Don't allow --dest to start with a dot (breaking)
* Rename the --dest flag to --link (old flag kept for compat)

See commit messages for more details.

### CODE REVIEW

File: cmd/git-sync/main.go [hunk 1]
```diff
@@ -305,16 +308,23 @@ func main() {
...
-	if strings.Contains(*flDest, "/") {
-		fmt.Fprintf(os.Stderr, "ERROR: --dest must be a leaf name, not a path\n")
+	if strings.Contains(*flLink, "/") {
+		fmt.Fprintf(os.Stderr, "ERROR: --link must not contain '/'\n")
 		pflag.Usage()
 		os.Exit(1)
 	}
+	if strings.HasPrefix(*flLink, ".") {
+		fmt.Fprintf(os.Stderr, "ERROR: --link must not start with '.'\n")
+		flag.Usage()
```
Comment (stp-ip) [inline cmd/git-sync/main.go:325]:
```suggestion
		pflag.Usage()
```

File: cmd/git-sync/main.go [hunk 2]
```diff
@@ -1043,10 +1053,10 @@ DESCRIPTION
...
     git-sync can pull one time, or on a regular interval.  It can read from the
     HEAD of a branch, from a git tag, or from a specific git hash.  It will only
-    re-pull if the target has changed in the remote repository.  When it
-    re-pulls, it updates the destination directory atomically.  In order to do
-    this, it uses a git worktree in a subdirectory of the --root and flips a
-    symlink.
+	re-pull if the target has changed in the remote repository.  When it
+	re-pulls, it updates the destination directory atomically.  In order to do
+	this, it uses a git worktree in a subdirectory of the --root and flips a
+	symlink.
```
Comment (stp-ip) [inline cmd/git-sync/main.go:1059]: Tab?

**Review by stp-ip — CHANGES_REQUESTED** [review]

File: cmd/git-sync/main.go (hunk 1 above)
Comment (thockin) [reply cmd/git-sync/main.go:325]: Good catch!

File: cmd/git-sync/main.go (hunk 2 above)
Comment (thockin) [reply cmd/git-sync/main.go:1059]: I will have to find some way to auto-prevent that...

**Comment by thockin** [comment]
re-push with those changes

File: cmd/git-sync/main.go (hunk 2 above)
Comment (stp-ip) [reply cmd/git-sync/main.go]: Go test checking for tabs in said variable?

**Review by stp-ip — APPROVED** [review]
/lgtm

**Pull request merged**